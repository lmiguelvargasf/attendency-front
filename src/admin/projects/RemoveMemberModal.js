import React from 'react'
import { Space, Modal, Select } from 'antd'
const { Option } = Select

const RemoveMemberModal = ({ state, handleOk, handleCancel, handleOnChange }) => {
  return (
    <Modal
      title='Remove Member'
      visible={state.visible}
      onOk={handleOk}
      confirmLoading={state.confirmLoading}
      onCancel={handleCancel}
    >
      {
        state.members.length > 0 ? (
          <Space>
            <strong>Member:</strong>
            <Select style={{ width: 150 }} onChange={handleOnChange} value={state.memberToRemove} data-testid='remove-member-select'>
              {state.members.map(member => (
                <Option key={member.key} value={member.key}>
                  {member.preferredName ? member.preferredName : `${member.firstName} ${member.lastName}`}
                </Option>
              ))}
            </Select>
          </Space>
        ) : (
          <span data-testid='no-members-remove'>There are no more members to remove from this project.</span>
        )
      }
    </Modal>
  )
}

export default RemoveMemberModal
