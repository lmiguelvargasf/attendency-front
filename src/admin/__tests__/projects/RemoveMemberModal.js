import React from 'react'
import { render, cleanup } from '@testing-library/react'
import RemoveMemberModal from '../../projects/RemoveMemberModal'

describe('RemoveMemberModal component', () => {
  afterEach(cleanup)

  describe('no members to be removed', () => {
    let component
    beforeEach(() => {
      const state = {
        visible: true,
        confirmLoading: false,
        members: [],
        memberToRemove: null
      }
      component = (
        <RemoveMemberModal state={state} handleOk={jest.fn()} handleCancel={jest.fn()} handleOnChange={jest.fn()} />
      )
    })

    it('matches snapshot', () => {
      const { asFragment } = render(component)
      expect(asFragment()).toMatchSnapshot()
    })

    it('displays text indicating no members to remove', () => {
      const { getByTestId, queryByTestId } = render(component)
      expect(getByTestId('no-members-remove')).toHaveTextContent('no more members to remove')
      expect(queryByTestId('remove-member-select')).toBeNull()
    })
  })

  describe('there are members to be removed', () => {
    let component
    beforeEach(() => {
      const state = {
        visible: true,
        confirmLoading: false,
        members: [
          { key: 1, preferredName: 'Alpha' },
          { key: 1, preferredName: 'Beta' },
          { key: 1, preferredName: 'Gamma' }
        ],
        memberToRemove: null
      }
      component = (
        <RemoveMemberModal state={state} handleOk={jest.fn()} handleCancel={jest.fn()} handleOnChange={jest.fn()} />
      )
    })

    it('matches snapshot', () => {
      const { asFragment } = render(component)
      expect(asFragment()).toMatchSnapshot()
    })

    it('displays select', () => {
      const { getByTestId, queryByTestId } = render(component)
      expect(getByTestId('remove-member-select')).not.toBeNull()
      expect(queryByTestId('no-members-remove')).toBeNull()
    })
  })
})
