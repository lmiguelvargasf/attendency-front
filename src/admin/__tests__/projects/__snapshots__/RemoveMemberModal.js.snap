// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RemoveMemberModal component no members to be removed matches snapshot 1`] = `<DocumentFragment />`;

exports[`RemoveMemberModal component there are members to be removed matches snapshot 1`] = `<DocumentFragment />`;
